### PR TITLE
Refactor lib [Part2]

### PR DESCRIPTION
**Pull request by shdehnavi**
### ☑️ Resolves

The required adjustments have been made to the following namespaces:
- `/lib/Command`
- `/lib/Controller`
- `/lib/Deck`
- `/lib/Events`

The improvements:

- Adding return types
- Adding types to properties
- Replacing `strpos` with `str_starts_with` to improve code readability

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not required
- [x] 📘 API documentation in `docs/` has been updated or is not required
- [x] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required
- [x] 🔖 Capability is added or not needed

### CODE REVIEW

File: lib/Events/ModifyParticipantEvent.php [hunk 1]
```diff
@@ -48,14 +48,14 @@ public function getParameter(): string {
 	/**
 	 * @return int|string|bool
 	 */
-	public function getNewValue() {
+	public function getNewValue(): bool|int|string {
```
Comment (nickvergessen) [inline lib/Events/ModifyParticipantEvent.php:51]: Unless the type is added to the constructor, I don't think we should type the getter.

File: lib/Events/SignalingEvent.php [hunk 2]
```diff
@@ -28,7 +28,7 @@
 
 class SignalingEvent extends ParticipantEvent {
 	/** @var mixed */
-	protected $session;
+	protected mixed $session;
```
Comment (nickvergessen) [inline lib/Events/SignalingEvent.php:31]:
```suggestion
	protected ?string $session;
```

But should also be enforced on the setter and getter then

File: lib/Events/ModifyRoomEvent.php [hunk 3]
```diff
@@ -45,14 +45,14 @@ public function getParameter(): string {
 	/**
 	 * @return int|string|bool
 	 */
-	public function getNewValue() {
+	public function getNewValue(): bool|int|string {
```
[the remark this answers is not in the document]
Comment (shdehnavi) [reply lib/Events/ModifyRoomEvent.php:48]: It's addressed in the next commit

File: lib/Events/ModifyParticipantEvent.php (hunk 1 above)
Comment (shdehnavi) [reply lib/Events/ModifyParticipantEvent.php:51]: It's addressed in the next commit

File: lib/Events/ModifyRoomEvent.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (nickvergessen) [reply lib/Events/ModifyRoomEvent.php:48]: Doesn't seem to be adjusted

File: lib/Events/SignalingEvent.php (hunk 2 above)
Comment (nickvergessen) [reply lib/Events/SignalingEvent.php]: Also was not done.

File: lib/Events/ModifyRoomEvent.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (shdehnavi) [reply lib/Events/ModifyRoomEvent.php]: Sorry my commit wasn't push, It's fixed now

File: lib/Events/SignalingEvent.php (hunk 2 above)
Comment (shdehnavi) [reply lib/Events/SignalingEvent.php]: Sorry my commit wasn't push, It's fixed now

**Comment by nickvergessen** [comment]
Tests are failing and types are still not correct.
For now I don't consider this worth it, but risky to break things,
so I will close this.

**Pull request closed without merging**